test/unit/BPool.t.sol

```diff
@@ -472,23 +630,94 @@ contract BPool_Unit_Gulp is BasePoolTest {
 }
 
 contract BPool_Unit_GetSpotPrice is BasePoolTest {
-  function test_Revert_NotBoundTokenIn() private view {}
+  function test_Revert_NotBoundTokenIn(address _tokenIn, address _tokenOut) public {
+    vm.expectRevert('ERR_NOT_BOUND');
+    bPool.getSpotPrice(_tokenIn, _tokenOut);
+  }
 
-  function test_Revert_NotBoundTokenOut() private view {}
+  function test_Revert_NotBoundTokenOut(address _tokenIn, address _tokenOut) public {
+    _setRecord(_tokenIn, BPool.Record({bound: true, index: 0, denorm: 0, balance: 0}));
 
-  function test_Returns_SpotPrice() private view {}
+    vm.expectRevert('ERR_NOT_BOUND');
+    bPool.getSpotPrice(_tokenIn, _tokenOut);
+  }
 
-  function test_Revert_Reentrancy() private view {}
+  function test_Returns_SpotPrice(
+    address _tokenIn,
+    address _tokenOut,
+    uint256 _tokenInBalance,
+    uint256 _tokenInDenorm,
+    uint256 _tokenOutBalance,
+    uint256 _tokenOutDenorm,
+    uint256 _swapFee
+  ) public {
+    vm.assume(_tokenIn != _tokenOut);
+    _assumeCalcSpotPrice(_tokenInBalance, _tokenInDenorm, _tokenOutBalance, _tokenOutDenorm, _swapFee);
+
+    _setRecord(_tokenIn, BPool.Record({bound: true, index: 0, denorm: _tokenInDenorm, balance: _tokenInBalance}));
+    _setRecord(_tokenOut, BPool.Record({bound: true, index: 0, denorm: _tokenOutDenorm, balance: _tokenOutBalance}));
+    _setSwapFee(_swapFee);
```

Comment:
I suggest introducing `happyPath` modifiers to the getter tests to favour normalization.